From: Yaniv Rosner <yanivr@broadcom.com>
Date: Wed, 4 Apr 2012 01:40:02 +0000
Subject: [PATCH 05/11] bnx2x: Clear MDC/MDIO warning message
Patch-mainline: v3.4-rc2
Git-commit: ca7b91bbd1f150216e6354cc20818aa993f331f2
References: bnc#769035

This patch clears a warning message of "MDC/MDIO access timeout" which may
appear when interface is loaded due to missing clock setting before resetting
the LED, and starting periodic function too early.

Signed-off-by: Yaniv Rosner <yanivr@broadcom.com>
Signed-off-by: Eilon Greenstein <eilong@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Yuval Mintz <yuvalmin@broadcom.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/bnx2x/bnx2x_cmn.c  |    1 -
 drivers/net/bnx2x/bnx2x_link.c |    2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/bnx2x/bnx2x_cmn.c
+++ b/drivers/net/bnx2x/bnx2x_cmn.c
@@ -1805,7 +1805,6 @@ int bnx2x_nic_load(struct bnx2x *bp, int
 		 * bnx2x_periodic_task().
 		 */
 		smp_mb();
-		queue_delayed_work(bnx2x_wq, &bp->period_task, 0);
 	} else
 		bp->port.pmf = 0;
 
--- a/drivers/net/bnx2x/bnx2x_link.c
+++ b/drivers/net/bnx2x/bnx2x_link.c
@@ -11902,10 +11902,10 @@ int bnx2x_link_reset(struct link_params
 	 * Hold it as vars low
 	 */
 	 /* clear link led */
+	bnx2x_set_mdio_clk(bp, params->chip_id, port);
 	bnx2x_set_led(params, vars, LED_MODE_OFF, 0);
 
 	if (reset_ext_phy) {
-		bnx2x_set_mdio_clk(bp, params->chip_id, port);
 		for (phy_index = EXT_PHY1; phy_index < params->num_phys;
 		      phy_index++) {
 			if (params->phy[phy_index].link_reset) {
